### PR TITLE
Cleanup of Nordic's BLE controller HAL and drivers

### PR DESCRIPTION
**Pull request by aescolar**
Use nrfx HAL instead of using directly the MDK registers structures to access
registers with sideeffects.
This allows to remove the current test hooks without affecting the testing.
+
Removed old test hooks.

### CODE REVIEW

File: drivers/clock_control/nrf5_power_clock.c [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #include <nrf_power.h>
 #include <drivers/clock_control/nrf5_clock_control.h>
 #endif
+#include "nrf_clock.h"
```
Comment (cvinayak) [inline drivers/clock_control/nrf5_power_clock.c]: Minor, could you move this above the #if clause, just to keep non-conditional includes together unless dependency requirements.

File: drivers/timer/nrf_rtc_timer.c [hunk 2]
```diff
@@ -552,11 +558,12 @@ void sys_clock_disable(void)
 
 	irq_disable(NRF5_IRQ_RTC1_IRQn);
 
-	SYS_CLOCK_RTC->EVTENCLR = RTC_EVTENCLR_COMPARE0_Msk;
-	SYS_CLOCK_RTC->INTENCLR = RTC_INTENCLR_COMPARE0_Msk;
+	nrf_rtc_event_disable(SYS_CLOCK_RTC, RTC_EVTENSET_COMPARE0_Msk);
+	nrf_rtc_int_disable(SYS_CLOCK_RTC, RTC_INTENSET_COMPARE0_Msk);
```
Comment (cvinayak) [inline drivers/timer/nrf_rtc_timer.c]: Why change CLR to SET msk?

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: drivers/clock_control/nrf5_power_clock.c (hunk 1 above)
Comment (aescolar) [reply drivers/clock_control/nrf5_power_clock.c]: done

File: drivers/timer/nrf_rtc_timer.c (hunk 2 above)
Comment (aescolar) [reply drivers/timer/nrf_rtc_timer.c]: No good reason apart from copy paste laziness and that they both are the same bit.
It is indeed better to use the CLR ones. So fixed it.

**Review by cvinayak — APPROVED** [review]
LGTM

**Comment by carlescufi** [comment]
Superseeds https://github.com/zephyrproject-rtos/zephyr/pull/7064

**Review by carlescufi — APPROVED** [review]
LGTM

**Pull request merged**